### PR TITLE
docs(conditions): added branch protection thread 

### PR DESCRIPTION
**Pull request by Syffe**
Following the injection of a new branch protection setting handler
in the engine, added a dedicated thread for branch protection settings.
Might be re-worked with more information.

### CODE REVIEW

File: docs/source/conditions.rst [hunk 1]
```diff
@@ -498,3 +498,54 @@ every status check that is expected, e.g.:
   that have ``build`` in their name (see point 1. and 2.above).
 
 Such conditions won't do what you want them to do.
+
+About Branch Protection
+~~~~~~~~~~~~~~~~~~~~~~~
+
+GitHub allows you to enable branch protection settings on your repository.
+For some of the branch protection parameters, listed as follows, Mergify is going to generate
+corresponding conditions in pull request and queue rules.
```
Comment (jd) [inline docs/source/conditions.rst:507]:
```suggestion
GitHub allows you to enable branch protection settings on your repository.
For some of the branch protection parameters, listed below, Mergify is going to generate
corresponding conditions in pull request and queue rules.
```

Where is the list? 😄

**Review by jd — COMMENTED** [review]
LGTM!

(don't forget to run `pngcrush` to reduce png image size)

File: docs/source/conditions.rst (hunk 1 above)
Comment (Syffe) [reply docs/source/conditions.rst:507]: the list is in the png image just below, it's a screen of the branch protection settings list in GitHub. I can eventually wright a list instead ! 

+ I modified the .png files and ran `pngcrush` on them as you said !

Comment (jd) [reply docs/source/conditions.rst:507]: I'd rather write it in text: images is not great to read, and you cannot search (C-f) in it :(

Comment (Syffe) [reply docs/source/conditions.rst]: Replaced the png with a list containing only the handled branch protection settings and changed some grammar like suggested

File: docs/source/conditions.rst [hunk 2]
```diff
@@ -498,3 +498,57 @@ every status check that is expected, e.g.:
...
+- ``Require conversation resolution before merging``
+
+Thus, enabling **"Require a pull request before merging"** and
+**"Require approvals"** will set the following condition:
+
+.. code-block:: yaml
+
+     conditions:
+       - #approved-reviews-by>=1
+       - #changes-requested-reviews-by=0
```
Comment (jd) [inline docs/source/conditions.rst:522]:
```suggestion
       - "#approved-reviews-by>=1"
       - "#changes-requested-reviews-by=0"
```

File: docs/source/conditions.rst [hunk 3]
```diff
@@ -498,3 +498,57 @@ every status check that is expected, e.g.:
...
+corresponding conditions in pull request and queue rules:
+
+- ``Require a pull request before merging`` with ``Require approvals``
+
+- ``Require status checks to pass before merging``
+
+- ``Require conversation resolution before merging``
+
+Thus, enabling **"Require a pull request before merging"** and
+**"Require approvals"** will set the following condition:
```
Comment (jd) [inline docs/source/conditions.rst:516]:
```suggestion
Thus, enabling ``Require a pull request before merging`` and
``Require approvals`` will set the following condition:
```

Maybe use the same formatting as above?

File: docs/source/conditions.rst [hunk 4]
```diff
@@ -498,3 +498,57 @@ every status check that is expected, e.g.:
...
+     conditions:
+        - and:
+          - or:
+            -check-success=continuous-integration/travis-ci
+            -check-neutral=continuous-integration/travis-ci
+            -check-skipped=continuous-integration/travis-ci
+          - or:
+            -check-success=continuous-integration/circle-ci
+            -check-neutral=continuous-integration/circle-ci
+            -check-skipped=continuous-integration/circle-ci
```
Comment (jd) [inline docs/source/conditions.rst:538]:
```suggestion
            - check-success=continuous-integration/travis-ci
            - check-neutral=continuous-integration/travis-ci
            - check-skipped=continuous-integration/travis-ci
          - or:
            - check-success=continuous-integration/circle-ci
            - check-neutral=continuous-integration/circle-ci
            - check-skipped=continuous-integration/circle-ci
```

File: docs/source/conditions.rst [hunk 5]
```diff
@@ -498,3 +498,57 @@ every status check that is expected, e.g.:
...
+
+            ...
+
+Enabling **"Require conversation resolution before merging"** will set the
+following condition:
```
Comment (jd) [inline docs/source/conditions.rst:543]:
```suggestion
Enabling ``Require conversation resolution before merging`` will set the
following condition:
```

File: docs/source/conditions.rst [hunk 6]
```diff
@@ -498,3 +498,57 @@ every status check that is expected, e.g.:
...
+
+.. code-block:: yaml
+
+     conditions:
+       - #review-threads-unresolved=0
```
Comment (jd) [inline docs/source/conditions.rst:548]:
```suggestion
       - "#review-threads-unresolved=0"
```

**Review by jd — CHANGES_REQUESTED** [review]

File: docs/source/conditions.rst (hunk 6 above)
Comment (Syffe) [reply docs/source/conditions.rst]: Done all the changes !

**Review by jd — APPROVED** [review]

**Pull request merged**